src/System.Text.Json/tests/Utf8JsonReaderTests.MultiSegment.cs

```diff
@@ -482,7 +484,7 @@ public static void CheckOnlyOneOfValueSpanOrSequenceIsSet(int testCase, string j
                     buffers[4] = dataUtf8.AsSpan(88, 2).ToArray();
                     sequence = BufferFactory.Create(buffers);
                     expectedHasValueSequenceSkip = new bool[] { false, false, false, true,  false, false };
-                    expectedHasValueSequenceAllow = new bool[] { false, false, false, true, true, true, false, false };
+                    expectedHasValueSequenceAllow = new bool[] { false, false, false, false, true, true, false, false };
```

Comment:
Why was this change necessary?

Reply:
At line [480](https://github.com/dotnet/corefx/pull/35934/files/4c9d01e68525cfaa487e4016e217af672bc54ab9#diff-297e5e4a465c2b75335c68e4098786d5R480), the buffer is being split right between `\r` and `\n` in the text `[// comment value\r\n\"value 2 across sequence\"` at line [440](https://github.com/dotnet/corefx/pull/35934/files/4c9d01e68525cfaa487e4016e217af672bc54ab9#diff-297e5e4a465c2b75335c68e4098786d5R440) with `\r` going with the comment text. Previously, comment delimiters and line separators were into comment _text_ and hence had a value sequence (hence was true). But with new changes that just return the comment text sans delimiters and line separators, the entire comment text fits in one span and hence `HasValueSequence` is false in this case. Hence the change.

Reply:
Ah yes, that makes sense.

Here's the split up input (where R => \n and N => \n)
```text
{"property name": [// comment valueR|
N"value 2 acr|
oss sequence"// another commen|
t valueRN|
]}
```

We should probably codify, in tests, what each of these splits mean (`|` indicates the split location):
`|// comment valueRN` => HasValueSequence = false
`/|/ comment valueRN` => HasValueSequence = false
`//| comment valueRN` => HasValueSequence = false**
`// |comment valueRN` => HasValueSequence = true
`// comment valu|eRN` => HasValueSequence = true
`// comment value|RN` => HasValueSequence = false**
`// comment valueR|N` => HasValueSequence = false
`// comment valueRN|` => HasValueSequence = false

** Let me know if you think these make sense.

Reply:
Will do these changes. Thanks!

Reply:
@ahsonkhan, I have done this in a bit different way to adjust with the existing test data.